shared/src/commonMain/kotlin/com/example/tasky/agenda/data/AgendaLocalDataSource.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.tasky.agenda.data
+
+import com.example.tasky.agenda.domain.model.Event
+import com.example.tasky.agenda.domain.model.Reminder
+import com.example.tasky.agenda.domain.model.Task
+import com.example.tasky.database.AppDatabase
+import com.example.tasky.database.database
+import com.example.tasky.database.mapper.toEventEntity
+import com.example.tasky.database.mapper.toReminderEntity
+import com.example.tasky.database.mapper.toTaskEntity
+import com.example.tasky.login.domain.manager.SessionManager
+
+class AgendaLocalDataSource(
+    private val appDatabase: AppDatabase = database,
+) {
+    fun deleteEvent(eventId: String) {
+        appDatabase.eventDao().delete(Event.EMPTY.copy(id = eventId).toEventEntity())
+    }
+
+    fun deleteTask(taskId: String) {
+        appDatabase.taskDao().delete(Task.EMPTY.copy(id = taskId).toTaskEntity())
+    }
+
+    fun deleteReminder(reminderId: String) {
+        appDatabase.reminderDao().delete(Reminder.EMPTY.copy(id = reminderId).toReminderEntity())
+    }
+
+    suspend fun upsertEvent(
+        event: Event,
+        isGoing: Boolean,
+    ) {
+        val userId = SessionManager.getUserId() ?: ""
+        val newAttendees = event.attendees.toMutableList()
+        val attendeeIndex = newAttendees.indexOfFirst { it.userId == userId }
+        if (attendeeIndex >= 0) {
+            val newAttendee = newAttendees[attendeeIndex].copy(isGoing = isGoing)
+            newAttendees[attendeeIndex] = newAttendee
+        }
+
+        appDatabase.eventDao().upsertEvent(
+            event.copy(attendees = newAttendees).toEventEntity(),
+        )
+    }
```

Comment:
This appears to be a side-effect that won’t be expected by anyone using the app 🤔 
I would be confused if I had a model and saved it to a database, and then I saw that the saved record had been altered. The creator of the event being always maked as going is business logic. That is because that rule can change with time. This logic makes more sense and is more transparent in your ViewModel.